test/compound/compound.test.ts

```diff
@@ -0,0 +1,127 @@
+import BN from "bn.js";
+import { u8aToHex } from "@polkadot/util";
+import { ApiPromise, WsProvider } from "@polkadot/api";
+import type { KeyringPair } from "@polkadot/keyring/types";
+import { chains } from "@oak-network/config";
+import { OakAdapter } from "@oak-network/adapter";
+import { rpc, types, runtime } from "@oak-network/types";
+import {
+  DEFAULT_TIMEOUT_INITIALIZE,
+  DEFAULT_TIMEOUT_PER_TEST,
+} from "../utils/constants";
+import { findEvent, getKeyringPair } from "../utils/helpFn";
+
+describe("compound", () => {
+  let keyringPair: KeyringPair | undefined;
+  let turingApi: ApiPromise | undefined;
+  let turingAdapter: OakAdapter | undefined;
+  beforeAll(async () => {
+    // Create keyringPair
+    keyringPair = await getKeyringPair();
+    const {
+      DevChains: { turingLocal: turingConfig },
+    } = chains;
+
+    // Initialize adapters
+    turingApi = await ApiPromise.create({
+      provider: new WsProvider(turingConfig.endpoint),
+      rpc,
+      runtime,
+      types,
+    });
+    turingAdapter = new OakAdapter(turingApi, turingConfig);
+    await turingAdapter.initialize();
+  }, DEFAULT_TIMEOUT_INITIALIZE);
+
+  afterAll(async () => {
+    await turingApi?.disconnect();
+  });
+
+  it(
+    "ensure-balance",
+    async () => {
+      expect(turingAdapter).toBeDefined();
+      expect(keyringPair).toBeDefined();
+      expect(keyringPair).toBeDefined();
+      await expect(
+        turingAdapter?.ensureBalance(
+          u8aToHex(keyringPair?.addressRaw),
+          new BN("10000000000"),
+        ),
+      ).resolves.toBeUndefined();
```

Comment:
and this verification code is so confusing, why it checks whether a function is undefined? After spending several minutes I finally understand what you want. Basically you want to make sure there’s no error thrown, but not to verify `toBeUndefined()`.

Please just update the function name to `isBalanceGreaterThan` or `hasEnoughBalance`, and return `true` or `false` instead.

Reply:
Yes, fixed it.